### PR TITLE
config: enable to set "MaxTPS" from environment variable

### PR DESCRIPTION
**Pull request by dragon3**
I'm opening this pull request because the same motivation as https://github.com/DataDog/docker-dd-agent/pull/291 . 
We are heavy users of APM and require a higher max_traces_per_second than what the default provides (10).

I know we can specify the value in datadog.yaml (tracer.sampler > max_traces_per_second), but it would be great if we can use an environment variable for it. Especially under container environment like Kubernetes, it's really useful and convenient.

### CODE REVIEW

**Comment by dragon3** [comment]
CI failed, but I'm not sure if it's related this pull request:
https://circleci.com/gh/DataDog/datadog-trace-agent/2385
```
=== RUN   TestMemHigh
FAIL	github.com/DataDog/datadog-trace-agent/watchdog	12.673s
```

File: config/merge_env.go [hunk 1]
```diff
@@ -11,21 +11,22 @@ import (
...
+	envHostname           = "DD_HOSTNAME"                  // agent hostname
+	envBindHost           = "DD_BIND_HOST"                 // statsd & receiver hostname
+	envReceiverPort       = "DD_RECEIVER_PORT"             // receiver port
+	envDogstatsdPort      = "DD_DOGSTATSD_PORT"            // dogstatsd port
+	envRemoteTraffic      = "DD_APM_NON_LOCAL_TRAFFIC"     // alow non-local traffic
+	envIgnoreResources    = "DD_IGNORE_RESOURCE"           // ignored resources
+	envLogLevel           = "DD_LOG_LEVEL"                 // logging level
+	envAnalyzedSpans      = "DD_APM_ANALYZED_SPANS"        // spans to analyze for transactions
+	envConnectionLimit    = "DD_CONNECTION_LIMIT"          // (deprecated) limit of unique connections
+	envMaxTracesPerSecond = "DD_APM_MAX_TRACES_PER_SECOND" // maximum limit to the total number of traces per second to sample (MaxTPS)
```
Comment (gbbr) [inline config/merge_env.go]: Let's call it "MaxTPS":
```go
envMaxTPS = "DD_MAX_TPS"
```

File: config/merge_env_test.go [hunk 2]
```diff
@@ -45,3 +46,22 @@ func TestAnalyzedSpansEnvConfigParsing(t *testing.T) {
 		assert.NotNil(err)
 	})
 }
+
+func TestLoadEnvMaxTracesPerSecond(t *testing.T) {
+	assert := assert.New(t)
+
+	t.Run("not exist DD_APM_MAX_TRACES_PER_SECOND envvar", func(t *testing.T) {
```
Comment (gbbr) [inline config/merge_env_test.go]: Please name this test "default":
```go
t.Run("default", ...
```
And the next one "env":
```go
t.Run("env", ...
```

File: config/merge_env_test.go [hunk 3]
```diff
@@ -45,3 +46,22 @@ func TestAnalyzedSpansEnvConfigParsing(t *testing.T) {
...
+		ac := &AgentConfig{}
```
Comment (gbbr) [inline config/merge_env_test.go]: Please use `New()` in both tests to get an `*AgentConfig` struct. It's a more real use-case.

File: config/merge_env_test.go [hunk 4]
```diff
@@ -45,3 +46,22 @@ func TestAnalyzedSpansEnvConfigParsing(t *testing.T) {
...
+		ac.loadEnv()
+		assert.EqualValues(0.0, ac.MaxTPS)
+	})
+
+	t.Run("exist DD_APM_MAX_TRACES_PER_SECOND envvar", func(t *testing.T) {
+		if err := os.Setenv("DD_APM_MAX_TRACES_PER_SECOND", "123.4"); err != nil {
+			t.Fatal(err)
+		}
```
Comment (gbbr) [inline config/merge_env_test.go]: You need to unset this too:
```go
defer os.Unsetenv("DD_MAX_TPS")
```

**Comment by gbbr** [comment]
P.S. Don't worry about CI, it's a separate issue that we are having with the `resource_class` setting in CircleCI. Should be fixed sometime soon.

**Comment by dragon3** [comment]
Thank you for the quick review and feedback!
I fixed all review comments. Can you please review it when you have a chance?

And I understood about the CI.
It would be great if this change is included in the next release.

Thanks!

**Comment by dragon3** [comment]
Thank you! 🎉

**Pull request merged**